lib/vfscore/automount.c

```diff
@@ -99,6 +145,16 @@ static int vfscore_volume_process_ukopts(struct vfscore_volume *vv)
 			o_next++;
 		}
 
+		/* First check is so we do not run `mkpath` on `/` */
+		if (vv->path[1] != '\0' && !strcmp(o_curr, "mkpath")) {
```

Comment:
I think it makes sense to add also a check that a mountpoint has always be specified with `vv->path[0] == '/'`. This should be independently checked of `mkpath`. If the path string does not start with `/` we should fail booting. Additionally, if the `/` check passed, we know it is safe to access `vv->path[1]`.

Reply:
Agreed, I am also going to add some more asserts in other places/functions.